pdns_exporter/pdns_exporter.py

```diff
@@ -0,0 +1,119 @@
+store JSON to files on disk. Data in JSON are in format to process in Passive DNS.
+"""
+
+parser = argparse.ArgumentParser(description='Module for exporting flow data to format for Passive DNS')
+parser.add_argument('-i',  "--ifcspec", help="select TRAP IFC specifier")
+args = parser.parse_args()
+
+context = pytrap.TrapCtx()
+context.init(['-i', args.ifcspec])
+
+context.setRequiredFmt(0, pytrap.FMT_UNIREC, "")
+unirec = pytrap.UnirecTemplate("ipaddr DST_IP,ipaddr SRC_IP,uint64 BYTES,uint64 LINK_BIT_FIELD,time TIME_FIRST, "\
+                               "time TIME_LAST,uint32 DNS_RR_TTL,uint32 PACKETS,uint16 DNS_ANSWERS,uint16 DNS_CLASS,"\
+                               "uint16 DNS_ID,uint16 DNS_PSIZE,uint16 DNS_QTYPE,uint16 DNS_RLENGTH,uint16 DST_PORT,"\
+                               "uint16 SRC_PORT,uint8 DIR_BIT_FIELD,uint8 DNS_DO,uint8 DNS_RCODE,"\
+                               "uint8 PROTOCOL,uint8 TCP_FLAGS,uint8 TOS,uint8 TTL,string DNS_NAME,bytes DNS_RDATA")
+
+records = list()
+CONTENT_THRESHOLD = 5000
+
+# ===== main lopp =====
+while True:
+    try:
+        data = context.recv()
+    except pytrap.FormatMismatch:
+        sys.stderr.write("Error: input data format mismatch.\n")
+        break
+    except pytrap.FormatChanged as e:
+        data = e.data
+        del(e)
+        pass
+    except pytrap.Terminated:
+        pass
+
+    if len(data) <= 1:
+        break
+
+    packets_n = unirec.get(data, "PACKETS")
+    # dns flow usually contains one packet
+    if packets_n == 1:
+        answers_n = unirec.get(data, "DNS_ANSWERS")
+        src_port = unirec.get(data, "SRC_PORT")
+        rcode = unirec.get(data, "DNS_RCODE")
+        # reponse message should have at least one answer and have 53 as source port
+        # and rcode with value 0 means no error happend
+        if answers_n >= 1 and src_port == 53 and rcode == 0:
+            # right now pdns support storage of A, AAAA and CNAME resource record types
+            rr_type = unirec.get(data, "DNS_QTYPE")
+            if rr_type not in (1, 5, 28):
+                continue
+
+            # For some flow the rcode contains value 1 but the response is type of CNAME
+            if rr_type == 1:
+                try:
+                    socket.inet_aton(unirec.get(data, "DNS_RDATA").decode('utf-8'))
+                except socket.error as e:
+                    continue
+
+            pdns_record = {
+                'response': unirec.get(data, "DNS_RDATA").decode('utf-8'),
+                'query': unirec.get(data, "DNS_NAME"),
+                'ttl': unirec.get(data, "DNS_RR_TTL"),
+                'type': unirec.get(data, "DNS_QTYPE"),
+                'time_first': unirec.get(data, "TIME_FIRST").toDatetime().isoformat(),
+                'time_last': unirec.get(data, "TIME_LAST").toDatetime().isoformat()
+            }
+
+            records.append(pdns_record)
+            if len(records) >= CONTENT_THRESHOLD:
```

Comment:
`CONTENT_THRESHOLD` seems something that can have some default value and can be redefined by an argument given by user.